backend/src/main/resources/db/migration/V1.4__fix_external_metadata_view.sql

```diff
@@ -0,0 +1,55 @@
+drop view if exists external_metadata_view cascade;
+
+create view external_metadata_view as
+select
+    cpd.accession,
+    cpd.version,
+    all_external_metadata.updated_metadata_at,
+    -- Combines metadata from preprocessed data with any external metadata updates
+    -- If there's no external metadata, just use the preprocessed data's metadata
+    -- If there is external metadata, merge it with preprocessed data (external takes precedence)
+    case 
+        when all_external_metadata.external_metadata is null then 
+            jsonb_build_object('metadata', (cpd.processed_data->'metadata'))
+        else 
+            jsonb_build_object(
+                'metadata', 
+                (cpd.processed_data->'metadata') || all_external_metadata.external_metadata
+            )
+    end as joint_metadata
+from
+    (
+        -- Get only the preprocessed data for the current pipeline version
+        select * from sequence_entries_preprocessed_data 
+        where pipeline_version = (select version from current_processing_pipeline)
```

Comment:
This is the key line here, need to filter _before_ join, as otherwise all processing versions end up in the view - Claude helped me figure this out!

Reply:
Yes - this was my finding as well - I wrote it in the issue